### ISSUE TITLE
Generalized Wasserstein Dice Loss

### ISSUE THREAD

**Issue by LucasFidon** [Project-MONAI/MONAI#601]
Add the Generalized Wasserstein Dice Loss (GWDL) loss function
https://arxiv.org/abs/1707.00478

The GWDL is a generalization of the Dice loss and the Generalized Dice loss that can tackle hierarchical classes and can take advantage of known relationships between classes.

I have already made a PyTorch implementation (including unit tests and an example) available here
https://github.com/LucasFidon/GeneralizedWassersteinDiceLoss

**Comment by arp95** [comment]
@wyli is this issue open? Can i work on it?

**Comment by wyli** [comment]
> @wyli is this issue open? Can i work on it?

yes, please go ahead, I'm assigning this ticket to you.

**Comment by arp95** [comment]
@Nic-Ma made changes as suggested in the review. do i need to work on tests or it would be done seperately?

**Comment by Nic-Ma** [comment]
Hi @arp95 ,

Actually, almost all the modules have the corresponding tests, so please add a unit test file to your PR referring to the test for regular DiceLoss: https://github.com/Project-MONAI/MONAI/blob/master/tests/test_dice_loss.py
Thanks.

**Comment by arp95** [comment]
Thanks @Nic-Ma . Will work on that.

**Comment by wyli** [comment]
resolved by https://github.com/Project-MONAI/MONAI/pull/744
